src/nvim/tui/term_input.inl

```diff
@@ -257,13 +258,11 @@ static TermInput *term_input_new(void)
   rv->paste_enabled = false;
   rv->in_fd = 0;
 
-  // Set terminal encoding based on environment(taken from libtermkey source
-  // code)
-  const char *e;
+  // Set terminal encoding based on environment
+  const char *ctype = setlocale(LC_CTYPE, NULL);
   int flags = 0;
-  if (((e = os_getenv("LANG")) || (e = os_getenv("LC_MESSAGES"))
-        || (e = os_getenv("LC_ALL"))) && (e = strchr(e, '.')) && e++ &&
-      (strcasecmp(e, "UTF-8") == 0 || strcasecmp(e, "UTF8") == 0)) {
+  if ( (ctype != NULL) && (ctype = strchr(ctype, '.')) && ctype++ &&
```

Comment:
Yes, I agree, but it's not clear that we're even going to keep this construct, as opposed to relying on `nl_langinfo`.